src/main/java/org/jitsi/videobridge/xmpp/config/Config.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright @ 2018 - present 8x8, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jitsi.videobridge.xmpp.config;
+
+import com.typesafe.config.*;
+import org.jitsi.utils.config.*;
+import org.jitsi.videobridge.util.config.*;
+import org.jitsi.xmpp.mucclient.*;
+
+import java.util.*;
+import java.util.stream.*;
+
+public class Config
+{
+    public static class XmppClientApiConfig
+    {
+        protected static class EnabledProperty extends AbstractConfigProperty<Boolean>
+        {
+            protected static final String propName = "videobridge.apis.xmpp-client.enabled";
+
+            private static EnabledProperty singleton = new EnabledProperty();
+
+            public EnabledProperty()
+            {
+                super(new JvbPropertyConfig<Boolean>()
```

Comment:
I would much rather these properties use a more declarative syntax. One shouldn't have to follow code just to understand the characteristics of a simple boolean property.

Reply:
Ok, I'm open to suggestions how to reimplement the config properties to achieve this.